File: S2M1/Ruby/carrera_carretera.rb
class CarreraCarretera < Carrera
    @@TASA = 0.1

    def self.TASA
        @@TASA
    end

    def run
        comenzar_carrera
        sleep(@@DURACION)
        @hebra.each(&:join)
        acabar_carrera
    end

    def initialize
        super
    end


    def comenzar_carrera
        puts "La carrera de carretera ha comenzado"
        super
    end

    def acabar_carrera
        puts "La carrera de carretera ha terminado"
    end

    public_class_method :new
end

File: S2M1/src/CarreraCarretera.java
public class CarreraCarretera extends Carrera implements Runnable {

    CarreraCarretera(int numberRace, ArrayList<_Bicicleta> bicicletas){
        super(numberRace, bicicletas);
    }

    public void startRace(){
        Ended = false;
        System.out.println("La carrera de carretera  " + numberRace +" ha comenzado");

    }

    public void endRace(){
        Ended = true;
        System.out.println("La carrera de carretera  " + numberRace +" ha terminaddo");
    }
}